mat-process-api.Tests/V1/Controllers/ProcessDataControllerTests.cs

```diff
@@ -73,12 +69,114 @@ public void given_a_processRef_when_getprocessdata_method_is_called_it_then_call
+            _mockValidatorGet.Setup(v => v.Validate(It.IsAny<GetProcessDataRequest>())).Returns(fakeValidationResult);
+
+            string processRef = faker.Random.String();
+
+            GetProcessDataRequest request = new GetProcessDataRequest() { processRef = processRef };
+
+            //act
+            var controllerResponse = _processDataController.GetProcessData(request);
+            var result = (ObjectResult)controllerResponse;
+            var resultContents = (IList<ValidationFailure>)result.Value;
+
+            //assert
+            Assert.NotNull(controllerResponse);
+            Assert.IsInstanceOf<BadRequestObjectResult>(result); // should be bad request, since validation fails.
+            Assert.NotNull(result);
+            Assert.IsInstanceOf<IList<ValidationFailure>>(resultContents);
+            Assert.NotNull(resultContents);
+
+            Assert.AreEqual(errorCount, resultContents.Count); // there should be 1 validation failure coming from the invalid guid
+            Assert.AreEqual(400, result.StatusCode); // expecting bad request result
+        }
+
+        [Test]
+        public void test_that_controller_returns_correct_message_and_status_code_when_no_document_was_found()
+        {
+            //Arrange
+            GetProcessDataRequest request = new GetProcessDataRequest
+            {
+                processRef = faker.Random.Guid().ToString()
+            };
+
+            var errorResponse = $"Document with reference {request.processRef} was not found in the database";
+            _mockUsecase.Setup(x => x.ExecuteGet(request)).Throws<DocumentNotFound>();
+            _mockValidatorGet.Setup(x => x.Validate(request)).Returns(new FV.ValidationResult());
+
+            //Act
+            var actualResponse = _processDataController.GetProcessData(request);
+            var result = (ObjectResult)actualResponse;
+            var resultContent = result.Value;
+
+            //Assert
+            Assert.NotNull(actualResponse);
+            Assert.NotNull(result);
+            Assert.NotNull(resultContent);
+            Assert.AreEqual(errorResponse, resultContent);
+            Assert.AreEqual(404, result.StatusCode);
+        }
+
+        [Test]
+        public void test_that_get_controller_returns_correct_500_status_code_when_an_error_has_occured()
+        {
+            //Arrange
+            GetProcessDataRequest request = new GetProcessDataRequest()
+            {
+                processRef = faker.Random.Guid().ToString()
+            };
+
+            _mockUsecase.Setup(x => x.ExecuteGet(request)).Throws<AggregateException>();
```

Comment:
Triangulation improvement possible after the merge though...

Reply:
Updated to use ErrorThrowerHelper